test/jit/test_pdt.py

```diff
@@ -0,0 +1,182 @@
+import os
+import sys
+import torch
+from torch.testing._internal.jit_utils import JitTestCase, make_global
+from torch.jit._monkeytype_config import _IS_MONKEYTYPE_INSTALLED
+from typing import List, Dict, Tuple  # noqa: F401
+
+# Make the helper files in test/ importable
+pytorch_test_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+sys.path.append(pytorch_test_dir)
+
+if not _IS_MONKEYTYPE_INSTALLED:
+    print("monkeytype is not installed. Skipping tests for Profile-Directed Typing ", file=sys.stderr)
+    JitTestCase = object  # type: ignore[misc, assignment] # noqa: F811
+
+if __name__ == "__main__":
+    raise RuntimeError(
+        "This test file is not meant to be run directly, use:\n\n"
+        "\tpython test/test_jit.py TESTNAME\n\n"
+        "instead."
+    )
+
+class TestPDT(JitTestCase):
```

Comment:
Have you tested whether using `torch.jit._script_pdt` as decorator on functions work?

Reply:
No I haven't. Let me give it a try.

Reply:
Created an issue for it.